src/main/java/mate/academy/spring/boot/service/BookServiceImpl.java

```diff
@@ -25,13 +33,10 @@ public Book save(Book book) {
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
-    }
-
-    @Override
-    public List<BookDto> getAll() {
-        throw new UnsupportedOperationException("Method not implemented");
+    public List<BookDto> findAll(Pageable pageable) {
+        return bookRepository.findAll(pageable).stream()
+                .map(bookMapper::toDto)
+                .toList();
```

Comment:
I think MapStruct supports mapping Page<> to List<>. like:
`List<BookDto> toDtos(Page<Book> pageable)`

Reply:
have you tried it?